src/newfight.cpp

```diff
@@ -1036,6 +1050,13 @@ void hit_with_multiweapon_toggle(struct char_data *attacker, struct char_data *v
         
         att->melee->power -= GET_IMPACT(def->ch) / 2;
       }
+      // Because we swap att and def pointers if defender wins the clash we need to make sure attacker gets proper values
+      // if they're using a ranged weapon in clash instead of setting their melee power to the damage code of the ranged
+      // weapon as it was happening.
+      else if (IS_RANGED(att->weapon)) {
+        att->melee->power = GET_STR(att->ch);
+        att->melee->damage_level = MODERATE;
```

Comment:
The main problem here was that the code went from this:
`if (GET_OBJ_RNUM(att->weapon) >= 0 && obj_index[GET_OBJ_RNUM(att->weapon)].wfunc == monowhip) {`

to this else block:
```
else {
        att->melee->power = GET_WEAPON_STR_BONUS(att->weapon) + GET_STR(att->ch);
        att->melee->power -= GET_IMPACT(def->ch);
      }```
      
This obviously overwrites those initially set values with the values of the current weapon. And this would not normally be a problem because we're attacking with a melee weapon BUT, a few lines above, we swap the pointers around if the defender wins the clash (which is clever in itself as it saves us rewriting a lot of code), which means that in that case, we're now doing this for the defender without any check if the weapon they're actually holding is a melee weapon or a gun. This results in defenders winning a clash with a ranged weapon using its damage code in the melee clash and a mob holding eg an AR in FA mode murdering you with dam code it shouldn't have (an adept with counter holding a canon yay).

I could just check if the weapon is ranged and if not leave the values as they are but thought this was cleaner to begin with.

Reply:
Ahh, got it. Works for me then, thanks!